operator/controllers/csiscaleoperator_controller.go

```diff
@@ -242,41 +234,30 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 
 		if err := r.deleteClusterRolesAndBindings(instance); err != nil {
-			message := "Failed to delete ClusterRoles and ClusterRolesBindings"
+			message := "Failed to delete the ClusterRoles and ClusterRoleBindings for the CSISCaleOperator instance " + instance.Name + ".\n" +
+				fmt.Sprintf("To get the list of the ClusterRoles and ClusterRoleBindings, use the selector as --selector='product=%s'", config.Product)
 			logger.Error(err, message)
-			// TODO: Add event.
-			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
-				Type:    string(config.StatusConditionSuccess),
-				Status:  metav1.ConditionFalse,
-				Reason:  string(csiv1.ResourceDeleteError),
-				Message: message,
-			})
+			SetStatusAndRaiseEvent(instance, r.Recorder, corev1.EventTypeWarning, string(config.StatusConditionSuccess),
+				metav1.ConditionFalse, string(csiv1.DeleteFailed), message,
+			)
 			return ctrl.Result{}, err
 		}
 
 		if err := r.deleteCSIDriver(instance); err != nil {
-			message := "Failed to delete CSIDriver"
+			message := fmt.Sprintf("Failed to delete the CSIDriver %s for the CSISCaleOperator instance %s", config.DriverName, instance.Name)
 			logger.Error(err, message)
-			// TODO: Add event.
-			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
-				Type:    string(config.StatusConditionSuccess),
-				Status:  metav1.ConditionFalse,
-				Reason:  string(csiv1.ResourceDeleteError),
-				Message: message,
-			})
+			SetStatusAndRaiseEvent(instance, r.Recorder, corev1.EventTypeWarning, string(config.StatusConditionSuccess),
+				metav1.ConditionFalse, string(csiv1.DeleteFailed), message,
+			)
 			return ctrl.Result{}, err
 		}
 
 		if err := r.removeFinalizer(instance); err != nil {
-			message := "Failed to remove Finalizer"
+			message := fmt.Sprintf("Failed to delete the finalizer %s for the CSISCaleOperator instance %s", config.CSIFinalizer, instance.Name)
 			logger.Error(err, message)
-			// TODO: Add event.
-			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
-				Type:    string(config.StatusConditionSuccess),
-				Status:  metav1.ConditionFalse,
-				Reason:  string(csiv1.ResourceUpdateError),
-				Message: message,
-			})
+			SetStatusAndRaiseEvent(instance, r.Recorder, corev1.EventTypeWarning, string(config.StatusConditionSuccess),
+				metav1.ConditionFalse, string(csiv1.DeleteFailed), message,
```

Comment:
DeleteFailed --> UpdateFailed
(as we are updating cso instance here while removing finalizer string)